lib/ecto/query/planner.ex

```diff
@@ -2207,8 +2207,8 @@ defmodule Ecto.Query.Planner do
       {{:ok, {:struct, _}}, {:fragment, _, _}} ->
         error!(query, "it is not possible to return a struct subset of a fragment")
 
-      {{:ok, {:struct, _}}, %Ecto.SubQuery{}} ->
-        error!(query, "it is not possible to return a struct subset of a subquery")
+      {{:ok, {:struct, fields}}, %Ecto.SubQuery{select: select}} ->
+        handle_subquery_struct(select, fields, ix, query)
```

Comment:
Given all functions seems to have a `select_` prefix, we should probably follow that here:

```suggestion
        select_subquery_struct(select, fields, ix, query)
```

Or `subquery_select_fields`.